### PR TITLE
[cli] Migrate withddl/workflow/healthstreamer flags to pflags

### LINKED ISSUE #10982: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tabletserver`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tabletserver/*.go
go/vt/vttablet/tabletserver/health_streamer.go: streamHealthBufferSize = flag.Uint("stream_health_buffer_size", 20, "max streaming health entries to buffer per streaming health client")
```

### LINKED ISSUE #10989: Switch flag definitions to be on pflag instead of flag in `package go/vt/withddl`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/withddl/*.go
go/vt/withddl/withddl_test.go:  flag.Parse() // Do not remove this comment, import into google3 depends on it
```

### LINKED ISSUE #10991: Switch flag definitions to be on pflag instead of flag in `package go/vt/workflow`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/workflow/*.go
go/vt/workflow/sleep_workflow.go:       subFlags := flag.NewFlagSet(sleepFactoryName, flag.ContinueOnError)
go/vt/workflow/workflow_test.go:        subFlags := flag.NewFlagSet(testWorkflowFactoryName, flag.ContinueOnError)
```

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

Migrates a few modules: withddl, workflow and health streamer from flag to pflag. Details in issues listed below.

## Related Issue(s)

Fixes #10989
Fixes #10991
Fixes #10982

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [X] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/workflow/sleep_workflow.go [hunk 1]
```diff
@@ -18,11 +18,12 @@ package workflow
...
 import (
 	"encoding/json"
-	"flag"
 	"fmt"
 	"sync"
 	"time"
 
+	"github.com/spf13/pflag"
+
 	"context"
```
Comment (ajm188) [inline go/vt/workflow/sleep_workflow.go:27]: can you move this up to the top? (this is an artifact that this used to be an `x/tools` package (so not with the stdlib imports) and someone did a mass-`sed` to replace them))

File: go/vt/workflow/workflow_test.go [hunk 2]
```diff
@@ -18,11 +18,12 @@ package workflow
...
-	"flag"
 	"fmt"
 	"strconv"
 	"sync"
 
+	"github.com/spf13/pflag"
+
 	"google.golang.org/protobuf/proto"
 
 	"context"
```
Comment (ajm188) [inline go/vt/workflow/workflow_test.go:29]: ditto

**Review by ajm188 — APPROVED** [review]
LGTM, two comments:

1. did you see my comment/question on https://github.com/vitessio/vitess/issues/10991#issuecomment-1211803637 ?
2. if we can't delete it, i don't care (much, but i _do_ still care a little) about the rest of the imports, but i do care about `context` ❤️

File: go/vt/workflow/sleep_workflow.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/vt/workflow/sleep_workflow.go]: > LGTM, two comments:
> 
> 1. did you see my comment/question on [#10991 (comment)](https://github.com/vitessio/vitess/issues/10991#issuecomment-1211803637) ?
> 2. if we can't delete it, i don't care (much, but i _do_ still care a little) about the rest of the imports, but i do care about `context` heart

Moved the context import. The deprecated flags can be removed when the refactor actually happens.

**Pull request merged**